apps/gk_field.c

```diff
@@ -55,10 +55,19 @@ gk_field_new(struct gkyl_gk *gk, struct gkyl_gyrokinetic_app *app)
       f->sheath_vals[2*j+1] = mkarr(app->use_gpu, 2*app->confBasis.num_basis, app->local_ext.volume);
     }
   } else {
+    struct gkyl_array* bmag_mid_host = app->gk_geom->bmag_mid;
+    if (app->use_gpu) {
+      bmag_mid_host = mkarr(false, 1, 1);
+      gkyl_array_copy(bmag_mid_host, app->gk_geom->bmag_mid);
+    }
+    double *bmag_mid_ptr = gkyl_array_fetch(bmag_mid_host, 0);
+    double bmag_fac = f->info.bmag_fac ? f->info.bmag_fac : bmag_mid_ptr[0];
```

Comment:
@akashukla I think we should change `bmag_fac` in the app (and input files if it's still in use) to bmag_ref. `_fac` is super vague. Or since we have `polarization_density` and this bmag_fac is specifically for the polarization, maybe it should be `polarization_bmag`